src/Sentry/Platforms/Native/UnmanagedHttpContent.cs

```diff
@@ -0,0 +1,68 @@
+using Sentry.Extensibility;
+
+namespace Sentry.Native;
+
+internal sealed class UnmanagedHttpContent : SerializableHttpContent
+{
+    private IntPtr _content;
+    private readonly int _length = 0;
+    private readonly IDiagnosticLogger? _logger;
+
+    public UnmanagedHttpContent(IntPtr content, int length, IDiagnosticLogger? logger)
+    {
+        _content = content;
+        _length = length;
+        _logger = logger;
+    }
+
+    ~UnmanagedHttpContent()
+    {
+        Dispose(false);
+    }
+
+    protected override async Task SerializeToStreamAsync(Stream stream, TransportContext? context)
+    {
+        try
```

Comment:
Oh ... one more thing:
To be on the safe side, we should throw an `ObjectDisposedException` in the `protected override` methods, in case `Dispose` has been called ... in our case when `_content == IntPtr.Zero` ... because we would otherwise get some random memory, perhaps even a `AccessViolationException`.

But again ... the type is `internal` so we can ensure we're using it correctly.
But perhaps to pay it forward to the next developer reading/changing this portion of the code.
We could do this in a follow-up as well ... as our current usage is correct.

Reply:
thanks for humoring me